apps/navstar_overlay/src/navstar_overlay_lashup_kv_listener.erl

```diff
@@ -0,0 +1,160 @@
+%% @private
+%% @doc
+%% This function is called by a gen_statem when it is about to
+%% terminate. It should be the opposite of Module:init/1 and do any
+%% necessary cleaning up. When it returns, the gen_statem terminates
+%% with Reason. The return value is ignored.
+%%
+%% @spec terminate(Reason, State, Data) -> void()
+%% @end
+%%--------------------------------------------------------------------
+-spec(terminate(Reason :: (normal | shutdown | {shutdown, term()} | term()),
+    State :: atom(), Data :: #data{}) -> term()).
+terminate(_Reason, _State, _Data) ->
+    ok.
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% Convert process state when code is changed
+%%
+%% @spec code_change(OldVsn, OldState, OldData, Extra) -> {state_functions, NewState, NewData}
+%% @end
+%%--------------------------------------------------------------------
+-spec(code_change(OldVsn :: term() | {down, term()}, OldState :: atom(),
+    OldData :: #data{}, Extra :: term()) ->
+    {ok, NewState :: atom(), NewData :: #data{}}).
+code_change(_OldVsn, OldState, OldData, _Extra) ->
+    {ok, OldState, OldData}.
+
+%%-------------------------------------------------------------------------------------------------------------
+%% State transition
+%% 
+%%                                                                lashup_event
+%%                                                                  +------+
+%%                                                                  |      |
+%%  +--------------+              +----------------+              +-+--------+                +-------------+
+%%  |              |              |                |              |          |                |             |
+%%  |              | lashup_event |                | lashup event |          |   timeout      |             |
+%%  | unconfigured +------------> |    applying    +------------> | batching +--------------> | configuring |
+%%  |              |              |  first_config  |              |          |                |             |
+%%  |              |              |                |              |          | <--------------+             |
+%%  |              |              |                |              |          |  lashup_event  |             |
+%%  +--------------+              +----------------+              +----------+                +-------------+
+%%
+%%--------------------------------------------------------------------------------------------------------------
+
+unconfigured(info, EventContent, Data) ->
+    lager:error("unconfigured"),
+    {next_state, applying_first_config, Data, {next_event, internal, EventContent}}.
+
+applying_first_config(internal, 
+            _EventContent = {lashup_kv_events, Event = #{key := [navstar, overlay, Subnet], ref := Reference}},
+            _Data = #data{ref = Ref}) when Ref == Reference, is_tuple(Subnet), tuple_size(Subnet) == 2 ->
```

Comment:
you don't actually need to do when Ref == Reference, you can do #data{ref = Reference}, and it'll match.